### PR TITLE
Implementation of "stop if true" conditional formatting feature.

### PR DESCRIPTION
**Pull request by f20**
This patch would fix issue #138.

### CODE REVIEW

**Comment by jmcnamara** [comment]
Once the changes are made do a git rebase or ammend and force push the changes back to the branch.

Thanks.

File: examples/conditional_format.pl [hunk 1]
```diff
@@ -88,6 +96,51 @@
 
 ###############################################################################
 #
+# Example 9 (uses the same data as example 1)
```
Comment (jmcnamara) [inline examples/conditional_format.pl]: * Move Example 9 after Example 8. 
* Drop "uses the same data as example 1" in this comment.
* All comments should end with a full stop.

File: examples/conditional_format.pl [hunk 2]
```diff
@@ -88,6 +96,51 @@
...
+#
+$caption = 'Cells with values >= 100 are in blue. '
```
Comment (jmcnamara) [inline examples/conditional_format.pl]: Update the caption to reflect what the example demonstrates.

File: examples/conditional_format.pl [hunk 3]
```diff
@@ -88,6 +96,51 @@
...
+  . 'Other cells with values >= 50 are in light red. '
+  . 'Values < 50 are in light green.';
+
+# Write the data.
+$worksheet9->write( 'A1', $caption );
+$worksheet9->write_col( 'B3', $data );
+
+# Write a conditional format over a range.
+# Use stopIfTrue to prevent previous formats from being used.
```
Comment (jmcnamara) [inline examples/conditional_format.pl]: Remember to change `stopIfTrue ` everywhere.

**Comment by jmcnamara** [comment]
Merged. Thanks.

**Comment by jmcnamara** [comment]
Is the `stop_if_true` criteria actually required in Excel 2997? In your example the same behaviour occurs with `stop_if_true = 0`.

The Office documentation on [Conditional Formatting Precedence](https://support.office.com/en-us/article/Manage-conditional-formatting-rule-precedence-063cde21-516e-45ca-83f5-8e8126076249) suggests that is mainly to support backward compatibility.

**Comment by f20** [comment]
I plead guilty… to guessing what the feature (which is prominent in Excel's user interface) did, and I guessed wrong so that my example failed to demonstrate anything.

Improved example here if you want it: 
https://github.com/jmcnamara/excel-writer-xlsx/compare/master...f20:stopIfTrue

**Comment by jmcnamara** [comment]
That's fine. I'll leave the example as it is.

If someone needs to use that feature they will probably know what they need to use it for. :-)

**Pull request merged**